hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -192,6 +237,158 @@ singlePartyCommitsFromExternal tracer workDir node@RunningNode{networkId} hydraS
  where
   RunningNode{nodeSocket} = node
 
+-- | Single hydra-node where the commit is done from an external UTxO owned by a
+-- script which requires providing script, datum and redeemer instead of
+-- signing the transaction.
+singlePartyCommitsFromExternalScript ::
```

Comment:
I would make this test a _BehaviorSpec_ test so that there is less plumbing involved and keep only a very basic test here.

Reply:
I am not sure we can test the same thing at this level. AFAIS `BehaviourSpec` is using `SimpleTx` and mocks the chain while in the end-to-end we have the api server available thus testing is easy there. We think there is no _behaviour_ to test here, we need to have the chain, api and the ledger available and I am not seeing the easy way to do it here.